app/models/host.rb

```diff
@@ -41,5 +39,21 @@ def filter_by_compliance_score(_filter, operator, score)
 
       { conditions: "hosts.id IN(#{ids.join(',')})" }
     end
+
+    def has_test_results(_filter, operator, value)
+      ids = Host.select do |h|
+        h.test_results.present? == ActiveModel::Type::Boolean.new.cast(value)
+      end
```

Comment:
Prefer SQL - against prod-sized data, `Host.search_for('has_test_results = true')` takes about 8 seconds :turtle: 

```suggestion
      ids = TestResult.pluck(:host_id)
```

or

```suggestion
      ids = Host.joins(:test_results).distinct.pluck(:id)
```

One is faster than the other:

```rb
[41] pry(main)> Set.new(Host.joins(:test_results).distinct.pluck(:id)) == Set.new(TestResult.pluck(:host_id))
   (15.8ms)  SELECT DISTINCT "hosts"."id" FROM "hosts" INNER JOIN "test_results" ON "test_results"."host_id" = "hosts"."id"
   (8.4ms)  SELECT "test_results"."host_id" FROM "test_results"
=> true
```